https://github.com/pagespeed/mod_pagespeed/issues/632

patterns used (vim):

update pattern to remove prefixes
:%s,\^\\#\([A-Z_]*\)$,s/&//,

update pattern to remove whole lines:
:%s,\^\\#\([A-Z_]*\)\.\*\\n,/^\\#\1/d,

update subst lines
:%s#-e "s@\$(\([A-Z_]*_PATTERN\))@@"#-e "$(\1)"#

--- modpagespeed-1.8.31.5/install/Makefile~	2014-12-14 16:05:55.000000000 +0200
+++ modpagespeed-1.8.31.5/install/Makefile	2014-12-14 16:12:55.900374377 +0200
@@ -190,176 +190,176 @@
 
 ifeq ($(STRESS_TEST),1)
   # remove prefix
-  STRESS_TEST_SED_PATTERN=^\#STRESS
+  STRESS_TEST_SED_PATTERN=s/^\#STRESS//
 else
   # remove whole line
-  STRESS_TEST_SED_PATTERN=^\#STRESS.*\n
+  STRESS_TEST_SED_PATTERN=/^\#STRESS/d
 endif
 
 ifeq ($(REWRITE_TEST),1)
   # remove prefix
-  REWRITE_TEST_SED_PATTERN=^\#REWRITE
+  REWRITE_TEST_SED_PATTERN=s/^\#REWRITE//
 else
   # remove whole line
-  REWRITE_TEST_SED_PATTERN=^\#REWRITE.*\n
+  REWRITE_TEST_SED_PATTERN=/^\#REWRITE/d
 endif
 
 ifeq ($(COVERAGE_TRACE_TEST),1)
   # remove coverage prefix
-  COVERAGE_TEST_SED_PATTERN=^\#COVERAGE
+  COVERAGE_TEST_SED_PATTERN=s/^\#COVERAGE//
 else
   # remove coverage lines
-  COVERAGE_TEST_SED_PATTERN=^\#COVERAGE.*\n
+  COVERAGE_TEST_SED_PATTERN=/^\#COVERAGE/d
 endif
 
 ifeq ($(PROXY_TEST),1)
   # remove prefix
-  PROXY_TEST_SED_PATTERN=^\#PROXY
+  PROXY_TEST_SED_PATTERN=s/^\#PROXY//
 else
   # remove whole line
-  PROXY_TEST_SED_PATTERN=^\#PROXY.*\n
+  PROXY_TEST_SED_PATTERN=/^\#PROXY/d
 endif
 
 ifeq ($(SLURP_TEST),1)
   # remove prefix
-  SLURP_TEST_SED_PATTERN=^\#SLURP
+  SLURP_TEST_SED_PATTERN=s/^\#SLURP//
 else
   # remove whole line
-  SLURP_TEST_SED_PATTERN=^\#SLURP.*\n
+  SLURP_TEST_SED_PATTERN=/^\#SLURP/d
 endif
 
 ifeq ($(SHARED_MEM_LOCK_TEST),1)
   # remove prefix
-  SHARED_MEM_LOCK_TEST_SED_PATTERN=^\#SHARED_MEM_LOCKS
+  SHARED_MEM_LOCK_TEST_SED_PATTERN=s/^\#SHARED_MEM_LOCKS//
 else
   # remove whole line
-  SHARED_MEM_LOCK_TEST_SED_PATTERN=^\#SHARED_MEM_LOCKS.*\n
+  SHARED_MEM_LOCK_TEST_SED_PATTERN=/^\#SHARED_MEM_LOCKS/d
 endif
 
 ifeq ($(MEMCACHED_TEST),1)
   # remove prefix
-  MEMCACHED_TEST_SED_PATTERN=^\#MEMCACHED
+  MEMCACHED_TEST_SED_PATTERN=s/^\#MEMCACHED//
 else
   # remove whole line
-  MEMCACHED_TEST_SED_PATTERN=^\#MEMCACHED.*\n
+  MEMCACHED_TEST_SED_PATTERN=/^\#MEMCACHED/d
 endif
 
 ifeq ($(IPRO_PRESERVE_COVERAGE_TEST),1)
   # remove prefix
-  IPRO_PRESERVE_COVERAGE_TEST_SED_PATTERN=^\#IPRO_PRESERVE_COVERAGE
+  IPRO_PRESERVE_COVERAGE_TEST_SED_PATTERN=s/^\#IPRO_PRESERVE_COVERAGE//
 else
   # remove whole line
-  IPRO_PRESERVE_COVERAGE_TEST_SED_PATTERN=^\#IPRO_PRESERVE_COVERAGE.*\n
+  IPRO_PRESERVE_COVERAGE_TEST_SED_PATTERN=/^\#IPRO_PRESERVE_COVERAGE/d
 endif
 
 ifeq ($(MEMCACHE_COVERAGE_TEST),1)
   # remove prefix
-  MEMCACHE_COVERAGE_TEST_SED_PATTERN=^\#MEMCACHE_COVERAGE
+  MEMCACHE_COVERAGE_TEST_SED_PATTERN=s/^\#MEMCACHE_COVERAGE//
 else
   # remove whole line
-  MEMCACHE_COVERAGE_TEST_SED_PATTERN=^\#MEMCACHE_COVERAGE.*\n
+  MEMCACHE_COVERAGE_TEST_SED_PATTERN=/^\#MEMCACHE_COVERAGE/d
 endif
 
 ifeq ($(PURGING_COVERAGE_TEST),1)
   # remove prefix
-  PURGING_COVERAGE_TEST_SED_PATTERN=^\#PURGING_COVERAGE
+  PURGING_COVERAGE_TEST_SED_PATTERN=s/^\#PURGING_COVERAGE//
 else
   # remove whole line
-  PURGING_COVERAGE_TEST_SED_PATTERN=^\#PURGING_COVERAGE.*\n
+  PURGING_COVERAGE_TEST_SED_PATTERN=/^\#PURGING_COVERAGE/d
 endif
 
 ifeq ($(IUR_COVERAGE_TEST),1)
   # remove prefix
-  IUR_COVERAGE_TEST_SED_PATTERN=^\#IUR_COVERAGE
+  IUR_COVERAGE_TEST_SED_PATTERN=s/^\#IUR_COVERAGE//
   # remove whole explicit domain authorization line
-  DOMAIN_AUTH_SED_PATTERN=^\#DOMAIN_AUTH_COVERAGE.*\n
+  DOMAIN_AUTH_SED_PATTERN=/^\#DOMAIN_AUTH_COVERAGE/d
 else
   # remove whole line
-  IUR_COVERAGE_TEST_SED_PATTERN=^\#IUR_COVERAGE.*\n
+  IUR_COVERAGE_TEST_SED_PATTERN=/^\#IUR_COVERAGE/d
   ifeq ($(COVERAGE_TRACE_TEST),1)
     # remove prefix for explicit domain authorization line
-    DOMAIN_AUTH_SED_PATTERN=^\#DOMAIN_AUTH_COVERAGE
+    DOMAIN_AUTH_SED_PATTERN=s/^\#DOMAIN_AUTH_COVERAGE//
   endif
 endif
 
 ifeq ($(SPELING_TEST),1)
   # remove prefix
-  SPELING_TEST_SED_PATTERN=^\#SPELING
+  SPELING_TEST_SED_PATTERN=s/^\#SPELING//
 else
   # remove whole line
-  SPELING_TEST_SED_PATTERN=^\#SPELING.*\n
+  SPELING_TEST_SED_PATTERN=/^\#SPELING/d
 endif
 
 ifeq ($(REWRITE_TEST),1)
   # remove prefix
-  REWRITE_TEST_SED_PATTERN=^\#REWRITE
+  REWRITE_TEST_SED_PATTERN=s/^\#REWRITE//
 else
   # remove whole line
-  REWRITE_TEST_SED_PATTERN=^\#REWRITE.*\n
+  REWRITE_TEST_SED_PATTERN=/^\#REWRITE/d
 endif
 
 ifeq ($(GZIP_TEST),1)
   # remove prefix
-  GZIP_TEST_SED_PATTERN=^\#GZIP
+  GZIP_TEST_SED_PATTERN=s/^\#GZIP//
 else
   # remove whole line
-  GZIP_TEST_SED_PATTERN=^\#GZIP.*\n
+  GZIP_TEST_SED_PATTERN=/^\#GZIP/d
 endif
 
 ifeq ($(EXPERIMENT_GA_TEST),1)
   # remove prefix
-  EXPERIMENT_GA_TEST_SED_PATTERN=^\#EXPERIMENT_GA
+  EXPERIMENT_GA_TEST_SED_PATTERN=s/^\#EXPERIMENT_GA//
 else
   # remove whole line
-  EXPERIMENT_GA_TEST_SED_PATTERN=^\#EXPERIMENT_GA.*\n
+  EXPERIMENT_GA_TEST_SED_PATTERN=/^\#EXPERIMENT_GA/d
 endif
 
 ifeq ($(EXPERIMENT_NO_GA_TEST),1)
   # remove prefix
-  EXPERIMENT_NO_GA_TEST_SED_PATTERN=^\#EXPERIMENT_NO_GA
+  EXPERIMENT_NO_GA_TEST_SED_PATTERN=s/^\#EXPERIMENT_NO_GA//
 else
   # remove whole line
-  EXPERIMENT_NO_GA_TEST_SED_PATTERN=^\#EXPERIMENT_NO_GA.*\n
+  EXPERIMENT_NO_GA_TEST_SED_PATTERN=/^\#EXPERIMENT_NO_GA/d
 endif
 
 ifeq ($(HTTPS_TEST),1)
   # remove prefix
-  HTTPS_TEST_SED_PATTERN=^\#HTTPS
+  HTTPS_TEST_SED_PATTERN=s/^\#HTTPS//
 else
   # remove whole line
-  HTTPS_TEST_SED_PATTERN=^\#HTTPS.*\n
+  HTTPS_TEST_SED_PATTERN=/^\#HTTPS/d
 endif
 
 ifeq ($(ALL_DIRECTIVES_TEST),1)
   # remove prefix
-  ALL_DIRECTIVES_TEST_SED_PATTERN=^\#ALL_DIRECTIVES
+  ALL_DIRECTIVES_TEST_SED_PATTERN=s/^\#ALL_DIRECTIVES//
 else
   # remove whole line
-  ALL_DIRECTIVES_TEST_SED_PATTERN=^\#ALL_DIRECTIVES.*\n
+  ALL_DIRECTIVES_TEST_SED_PATTERN=/^\#ALL_DIRECTIVES/d
 endif
 
 ifeq ($(PER_VHOST_STATS_TEST),1)
   # remove prefix
-  PER_VHOST_STATS_TEST_SED_PATTERN=^\#PER_VHOST_STATS
+  PER_VHOST_STATS_TEST_SED_PATTERN=s/^\#PER_VHOST_STATS//
 else
   # remove whole line
-  PER_VHOST_STATS_TEST_SED_PATTERN=^\#PER_VHOST_STATS.*\n
+  PER_VHOST_STATS_TEST_SED_PATTERN=/^\#PER_VHOST_STATS/d
 endif
 
 ifeq ($(NO_PER_VHOST_STATS_TEST),1)
   # remove prefix
-  NO_PER_VHOST_STATS_TEST_SED_PATTERN=^\#NO_PER_VHOST_STATS
+  NO_PER_VHOST_STATS_TEST_SED_PATTERN=s/^\#NO_PER_VHOST_STATS//
 else
   # remove whole line
-  NO_PER_VHOST_STATS_TEST_SED_PATTERN=^\#NO_PER_VHOST_STATS.*\n
+  NO_PER_VHOST_STATS_TEST_SED_PATTERN=/^\#NO_PER_VHOST_STATS/d
 endif
 
 ifeq ($(STATS_LOGGING_TEST),1)
   # remove prefix
-  STATS_LOGGING_TEST_SED_PATTERN=^\#STATS_LOGGING
+  STATS_LOGGING_TEST_SED_PATTERN=s/^\#STATS_LOGGING//
 else
   # remove whole line
-  STATS_LOGGING_TEST_SED_PATTERN=^\#STATS_LOGGING.*\n
+  STATS_LOGGING_TEST_SED_PATTERN=/^\#STATS_LOGGING/d
 endif
 
 # Note that the quoted sed replacement for APACHE_SLURP_DIR_COMMAND is because
@@ -381,27 +381,27 @@
 	    -e "s@# ModPagespeedSlurpReadOnly on@$(APACHE_SLURP_READ_ONLY_COMMAND)@g" \
 	    -e "s|@@TMP_SLURP_DIR@@|$(TMP_SLURP_DIR)|g" \
 	    -e "s|@@MEMCACHED_PORT@@|$(MEMCACHED_PORT)|g" \
-	    -e "s@$(STRESS_TEST_SED_PATTERN)@@" \
-	    -e "s@$(REWRITE_TEST_SED_PATTERN)@@" \
-	    -e "s@$(COVERAGE_TEST_SED_PATTERN)@@" \
-	    -e "s@$(PROXY_TEST_SED_PATTERN)@@" \
-	    -e "s@$(SLURP_TEST_SED_PATTERN)@@" \
-	    -e "s@$(SHARED_MEM_LOCK_TEST_SED_PATTERN)@@" \
-	    -e "s@$(SPELING_TEST_SED_PATTERN)@@" \
-	    -e "s@$(MEMCACHED_TEST_SED_PATTERN)@@" \
-	    -e "s@$(IPRO_PRESERVE_COVERAGE_TEST_SED_PATTERN)@@" \
-	    -e "s@$(MEMCACHE_COVERAGE_TEST_SED_PATTERN)@@" \
-	    -e "s@$(PURGING_COVERAGE_TEST_SED_PATTERN)@@" \
-	    -e "s@$(IUR_COVERAGE_TEST_SED_PATTERN)@@" \
-	    -e "s@$(DOMAIN_AUTH_SED_PATTERN)@@" \
-	    -e "s@$(GZIP_TEST_SED_PATTERN)@@" \
-	    -e "s@$(HTTPS_TEST_SED_PATTERN)@@" \
-	    -e "s@$(EXPERIMENT_GA_TEST_SED_PATTERN)@@" \
-	    -e "s@$(EXPERIMENT_NO_GA_TEST_SED_PATTERN)@@" \
-	    -e "s@$(ALL_DIRECTIVES_TEST_SED_PATTERN)@@" \
-	    -e "s@$(PER_VHOST_STATS_TEST_SED_PATTERN)@@" \
-	    -e "s@$(NO_PER_VHOST_STATS_TEST_SED_PATTERN)@@" \
-	    -e "s@$(STATS_LOGGING_TEST_SED_PATTERN)@@" \
+	    -e "$(STRESS_TEST_SED_PATTERN)" \
+	    -e "$(REWRITE_TEST_SED_PATTERN)" \
+	    -e "$(COVERAGE_TEST_SED_PATTERN)" \
+	    -e "$(PROXY_TEST_SED_PATTERN)" \
+	    -e "$(SLURP_TEST_SED_PATTERN)" \
+	    -e "$(SHARED_MEM_LOCK_TEST_SED_PATTERN)" \
+	    -e "$(SPELING_TEST_SED_PATTERN)" \
+	    -e "$(MEMCACHED_TEST_SED_PATTERN)" \
+	    -e "$(IPRO_PRESERVE_COVERAGE_TEST_SED_PATTERN)" \
+	    -e "$(MEMCACHE_COVERAGE_TEST_SED_PATTERN)" \
+	    -e "$(PURGING_COVERAGE_TEST_SED_PATTERN)" \
+	    -e "$(IUR_COVERAGE_TEST_SED_PATTERN)" \
+	    -e "$(DOMAIN_AUTH_SED_PATTERN)" \
+	    -e "$(GZIP_TEST_SED_PATTERN)" \
+	    -e "$(HTTPS_TEST_SED_PATTERN)" \
+	    -e "$(EXPERIMENT_GA_TEST_SED_PATTERN)" \
+	    -e "$(EXPERIMENT_NO_GA_TEST_SED_PATTERN)" \
+	    -e "$(ALL_DIRECTIVES_TEST_SED_PATTERN)" \
+	    -e "$(PER_VHOST_STATS_TEST_SED_PATTERN)" \
+	    -e "$(NO_PER_VHOST_STATS_TEST_SED_PATTERN)" \
+	    -e "$(STATS_LOGGING_TEST_SED_PATTERN)" \
 		$^ > $@
 	! grep '@@' $@  # Make sure we don't have any remaining @@variables@@
 
